src/protocol.ts

```diff
@@ -190,6 +215,53 @@ export class Protocol {
     return this._collateralSupply;
   }
 
+  /**
+   * Fetches and stores current PSM TVL.
+   * @returns Fetched PSM TVL.
+   */
+  async fetchPsmTvl(): Promise<PsmTvlData | null> {
+    if (RaftConfig.networkId !== 1) {
+      console.warn('PSM TVL is only available on mainnet');
+
+      this._psmTvl = {
+        daiLocked: new Decimal(0),
+        usdValue: new Decimal(0),
+      };
+
+      return this._psmTvl;
+    }
+
+    const chaiToken = ChaiToken__factory.connect(CHAI_TOKEN_ADDRESS, this.provider);
+
+    const chaiPotAddress = await chaiToken.pot();
```

Comment:
Maybe we can store it as const if it is not changing, just to avoid extra fetching.

Reply:
I'm not sure if it's changing, I check chai application code and they are doing it this way, maybe because it can potentially change.